### PR TITLE
Added support for Spring Cloud CircuitBreaker

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
fixes gh-279

### CODE REVIEW

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+				return false;
+			}
+		}
+		else if ("hashCode".equals(method.getName())) {
+			return hashCode();
+		}
+		else if ("toString".equals(method.getName())) {
+			return toString();
+		}
+		String circuitName = this.feignClientName + "_" + method.getName();
```
Comment (marcingrzejszczak) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java:78]: Is this a good name?

File: spring-cloud-openfeign-dependencies/pom.xml [hunk 2]
```diff
@@ -50,11 +50,6 @@
 				<artifactId>feign-httpclient</artifactId>
 				<version>${feign.version}</version>
 			</dependency>
-			<dependency>
```
Comment (marcingrzejszczak) [inline spring-cloud-openfeign-dependencies/pom.xml:53]: This is a breaking change!

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java (hunk 1 above)
Comment (OlgaMaciaszek) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java:78]: Seems ok to me. @ryanjbaxter wdyt?

File: spring-cloud-openfeign-dependencies/pom.xml (hunk 2 above)
Comment (OlgaMaciaszek) [reply spring-cloud-openfeign-dependencies/pom.xml:53]: Let's keep it in `master` but make sure we don't have it when we add these changes to `2.2.x`.

**Review by OlgaMaciaszek — APPROVED** [review]
Looks good to me. One thing to keep in mind is that these changes should be added to Hoxton (minus the breaking one) in an opt-in fashion (Hystrix used by default, possibility to switch to using CB).

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java (hunk 1 above)
Comment (ryanjbaxter) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java:78]: Documenting the name will be important.  Users need to know the circuit breaker name in order to customize it if they want https://docs.spring.io/spring-cloud-circuitbreaker/docs/1.0.4.RELEASE/reference/html/#specific-circuit-breaker-configuration

**Review by ryanjbaxter — APPROVED** [review]

**Comment by ryanjbaxter** [comment]
Back porting this to Hoxton is going to be messy, but doable I suppose

**Comment by marcingrzejszczak** [comment]
Closed via https://github.com/spring-cloud/spring-cloud-openfeign/commit/63a524843d3cfb9c6c1de45ca6cff5e1ea6e6a26

**Comment by marcingrzejszczak** [comment]
@ryanjbaxter https://github.com/spring-cloud/spring-cloud-openfeign/commit/63a524843d3cfb9c6c1de45ca6cff5e1ea6e6a26#diff-b258ace955d38ee849bf3d4a8e0afdb84712346131522e39bb4782b1cffc8506R332 - added such text

**Pull request closed without merging**